clock_exercise/AlarmClockListPage.qml

```diff
@@ -0,0 +1,93 @@
+import QtQuick 2.0
+
+Rectangle {
+    id: alarmClockList
+
+    signal back
+
+    color: "#151B2E"
+
+    BackButton {
+        x: 28
+        y: 20
+        onBack: {
+            clockManager.alarmItemModel.setAllNotSelected()
+            alarmClockList.back()
+        }
+    }
+
+    TitleOfThePage {
+        y: 40
+        anchors.horizontalCenter: parent.horizontalCenter
+        text: qsTr("Alarm clock list")
+    }
+
+    LineUnderTitleOfThePage {
```

Comment:
Il nome scelto per questa componente mi insospettisce un po', mi spiego meglio: quando noi spezziamo del codice QML per creare una nuova componente lo facciamo per una serie di motivi, tra cui rendere il codice più modulare, ovvero riutilizzabile anche in contesti diversi da quello in cui è nato.
Leggendo il nome capisco che questa componente è pensata solo ed esclusivamente per essere usata sotto il titolo della pagina, al che mi sorge spontanea una domanda: ma se questa componente è pensata solo ed esclusivamente per stare sotto il titolo della pagina, perchè non è dentro la componente `TitleOfThePage`?

Una linea generale quando si nominano le componenti, è quella di dare un nome basandosi sull'aspetto della componente piuttosto che in base all'utilizzo che ne stai facendo (anche se talvolta anche l'utilizzo è importante per il naming). Tendenzialmente, più una componente è semplice, più è facile evitare di riferisi all'utilizzo che ne fai per nominarla.
`TimerPage` ad esempio è una componente complessa, e inevitabilmente ne descriviamo l'utilizzo quando la nominiamo, e va bene così.
In questo caso però trattandosi di un semplice `Rectangle` con dei default su `width`, `height` e `color`, possiamo trovare un nome più adeguato.
Nei progetti in cui ho lavorato, ho sempre trovato questo component chiamato `Separator`.

Reply:
Potrebbe forse anche essere accorpato al titolo della pagina in un unico componente `PageHeader` o semplicemente `Header`

Reply:
Non tutti gli header sono uguali, quello della main page ha le iconcine dell'alarm list e del timer, gli altri hanno il tasto back, se faccio un componente header devo mettere solo il titolo e la linea, ma a questo punto sarebbe fuorviante, perché non è un vero hader, è un pezzo di header.